app/src/main/java/com/andrew/tasky/agenda/data/AgendaApi.kt

```diff
@@ -0,0 +1,53 @@
+package com.andrew.tasky.agenda.data
+
+import com.andrew.tasky.agenda.util.Event
+import java.util.TimeZone
+import okhttp3.MultipartBody
+import retrofit2.Response
+import retrofit2.http.DELETE
+import retrofit2.http.GET
+import retrofit2.http.Multipart
+import retrofit2.http.POST
+import retrofit2.http.PUT
+import retrofit2.http.Part
+import retrofit2.http.Query
+
+interface AgendaApi {
+
+    @GET("/agenda")
+    suspend fun getAgendaItems(
+        @Query("timezone")
+        timeZone: TimeZone,
```

Comment:
Have you tried if this works? Will it automatically serialize the `TimeZone` class to a string?